src/rules/prefer-importing-jest-globals.ts

```diff
@@ -0,0 +1,189 @@
+        const reportingNode = jestFunctionsToReport[0].head.node;
+
+        const jestFunctionsToImportFormatted = jestFunctionsToImport.join(', ');
+
+        const isModule = context.parserOptions.sourceType === 'module';
+
+        context.report({
+          node: reportingNode,
+          messageId: 'preferImportingJestGlobal',
+          data: { jestFunctions: jestFunctionsToImportFormatted },
+          fix(fixer) {
+            const sourceCode = getSourceCode(context);
+            const [firstNode] = sourceCode.ast.body;
+
+            // check if "use strict" directive exists
+            if (
+              firstNode.type === AST_NODE_TYPES.ExpressionStatement &&
+              firstNode.expression.type === AST_NODE_TYPES.Literal &&
+              firstNode.expression.value === 'use strict'
+            ) {
+              return fixer.insertTextAfter(
+                firstNode,
+                `\n${createFixerImports(isModule, jestFunctionsToImport)}`,
+              );
+            }
+
+            const importNode = sourceCode.ast.body.find(
+              node =>
+                node.type === AST_NODE_TYPES.ImportDeclaration &&
+                node.source.value === '@jest/globals',
+            );
+
+            if (
+              importNode &&
+              importNode.type === AST_NODE_TYPES.ImportDeclaration
+            ) {
+              const existingImports = importNode.specifiers.reduce<string[]>(
+                (imports, specifier) => {
+                  /* istanbul ignore else */
+                  if (
+                    specifier.type === AST_NODE_TYPES.ImportSpecifier &&
+                    specifier.imported?.name
+                  ) {
+                    imports.push(specifier.imported.name);
+                  }
+
+                  return imports;
+                },
+                [] as string[],
+              );
+
+              const allImports = [
+                ...new Set([...existingImports, ...jestFunctionsToImport]),
+              ];
+
+              return fixer.replaceText(
+                importNode,
+                createFixerImports(isModule, allImports),
+              );
+            }
+
+            const requireNode = sourceCode.ast.body.find(
+              node =>
+                node.type === AST_NODE_TYPES.VariableDeclaration &&
+                node.declarations.some(
+                  declaration =>
+                    declaration.init &&
+                    (declaration.init as any).callee &&
+                    (declaration.init as any).callee.name === 'require' &&
+                    (declaration.init as any).arguments?.[0]?.type ===
+                      'Literal' &&
+                    (declaration.init as any).arguments?.[0]?.value ===
+                      '@jest/globals',
+                ),
+            );
+
+            if (
+              requireNode &&
+              requireNode.type === AST_NODE_TYPES.VariableDeclaration
+            ) {
```

Comment:
```suggestion
            if (requireNode?.type === AST_NODE_TYPES.VariableDeclaration) {
```

Reply:
Also this is the much longer codepath - can we flip and do `!==` here instead?

Reply:
changed